src/inventory_ui.cpp

```diff
@@ -836,6 +866,8 @@ void inventory_column::prepare_paging( const std::string &filter )
             from = to;
         }
     }
+    order_by_parent();
+    order_by_category();
```

Comment:
Here's a problem: Line 836-867 still sorts only once every UI session, so if you have favorite items in your inventory, the entry order still changes when you open and close the inventory item menu. However if you remove the `entries_unfiltered.empty()` check and let it sort every screen reinitialization, it would cause items to jump around when you favorite items (because favoriting items may cause the required screen width to change, so a reinitialization call is required, and then the items are re-sorted according to the favorite status). So #44603 still needs to be addressed.

Also, when resizing the game window, items in the worn items column seems to change order randomly.